### PR TITLE
fix(patch): cover case where patch is used to create an entity

### PR DESCRIPTION
**Pull request by RyanHolstien**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/entity/AspectUtils.java [hunk 1]
```diff
@@ -73,9 +73,10 @@ private static MetadataChangeProposal getProposalFromAspect(String aspectName, R
     try {
       MetadataChangeProposal proposal = original.copy();
       GenericAspect genericAspect = GenericRecordUtils.serializeAspect(aspect);
-      // Set UPSERT changetype here as additional changes being added should always be
-      // done in UPSERT mode even for patches
-      // proposal.setChangeType(ChangeType.UPSERT);
+      // Additional changes should never be set as PATCH, if a PATCH is coming across it should be an UPSERT
+      if (ChangeType.PATCH.equals(proposal.getChangeType())) {
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/AspectUtils.java:77]: Looks good.

**Review by jjoyce0510 — APPROVED** [review]
LGTM

File: metadata-io/src/main/java/com/linkedin/metadata/entity/AspectUtils.java (hunk 1 above)
Comment (jjoyce0510) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/AspectUtils.java:77]: Do we have unit tests for this class?

Comment (RyanHolstien) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/AspectUtils.java:77]: Not directly, but it does get used on the OpenAPI controllers so it does get executed in tests

File: metadata-io/src/test/java/com/linkedin/metadata/AspectUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import com.linkedin.mxe.MetadataChangeProposal;
+import io.ebean.EbeanServer;
+import java.util.List;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.*;
+
+
+public class AspectUtilsTest {
```
Comment (jjoyce0510) [inline metadata-io/src/test/java/com/linkedin/metadata/AspectUtilsTest.java:28]: Thank you Ryan!

**Pull request merged**